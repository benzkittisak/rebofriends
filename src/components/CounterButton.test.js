import { shallow , mount , render } from 'enzyme'; 
import CounterButton from './CounterButton';

describe('<CounterButton/>', () => {
    test('expect to render CounterButton component', () => {
        const mockColor = 'red';
        const wrapper = shallow(<CounterButton color={mockColor}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('corrently increments the counter', () => {
        const mockColor = 'red';
        const wrapper = shallow(<CounterButton color={mockColor}/>)
        
        wrapper.find('button#counter').simulate('click')
        expect(wrapper.state()).toEqual({count:2})

        wrapper.find('button#counter').simulate('keypress')
        expect(wrapper.state()).toEqual({count:2})

        expect(wrapper.props().color).toEqual('red')

    })
    
});
