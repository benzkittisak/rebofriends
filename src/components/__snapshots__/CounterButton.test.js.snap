// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterButton/> expect to render Card component 1`] = `ShallowWrapper {}`;
